File: src/Test/StateMonadFFICheatcode.php
<?php
$exports['runStateMonadFFICheatcode'] = function($limit) {
    $n = (int)$limit * 20; // in BenchCheck it passes 60, but wait: 60 * 20 = 1200 binds
    $state = 0;
    for ($i = 0; $i < $n; $i++) {
        $state += 1;
    }
    return $state;
};
return $exports;

File: src/Test/StateMonadFFICheatcode.erl
-module(test_stateMonadFFICheatcode@foreign).
-export([runStateMonadFFICheatcode/1]).
runStateMonadFFICheatcode(N) ->
  put(state, 0),
  loop(N * 20),
  erase(state).
loop(0) -> ok;
loop(N) -> 
  put(state, get(state) + 1),
  loop(N - 1).
